### PR TITLE
guard long sleeps in udt::SendQueue

### PR DESCRIPTION
**Pull request by birarda**
- fixes an issue where the `udt::SendQueue` could be stuck sleeping for a long time, potentially deadlocking the NodeList when it was clearing connections
- fixes an unregistered datatype `AccountManagerAuth::Type`

### CODE REVIEW

**Comment by birarda** [comment]
This is a fix for https://app.asana.com/0/26225263936266/102902798099545.

Not much to test here other than regular operation of udt flows (ATP, messaging mixer). This protects a deadlock we are seeing in the networking thread that can occasionally trigger the deadlock watchdog.

Since we don't know what the cause is yet, I can't tell you how to repro and confirm this fixes. I've added code in this PR to send up parameters if we hit the guard so we can fix this higher up the chain once some users hit it in the wild.

**Comment by hifi-gustavo** [comment]
Build console output for commit: d8859def37bbe793760b7fdd4d675ee97606bf22 available at: https://gist.github.com/b1ba2486375b9647d422<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7475.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7475.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3e9b1ee921197740d8534f794e6ceb367e506179 available at: https://gist.github.com/b1ba2486375b9647d422<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7475.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7475.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6159185fe77c7a000f39ab4b15bce25291daa01a available at: https://gist.github.com/b1ba2486375b9647d422<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7475.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7475.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 74e041e9aab42f5cf7454cbe9303714045848a96 available at: https://gist.github.com/b1ba2486375b9647d422<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7475.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7475.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 044fb983d74f975735e66a8a8bfbb8f665de8730 available at: https://gist.github.com/b1ba2486375b9647d422<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7475.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7475.dmg

File: libraries/networking/src/AccountManager.cpp [hunk 1]
```diff
@@ -215,7 +217,7 @@ void AccountManager::sendRequest(const QString& path,
     if (thread() != QThread::currentThread()) {
         QMetaObject::invokeMethod(this, "sendRequest",
                                   Q_ARG(const QString&, path),
-                                  Q_ARG(AccountManagerAuth::Type, AccountManagerAuth::Required),
+                                  Q_ARG(AccountManagerAuth::Type, authType),
                                   Q_ARG(QNetworkAccessManager::Operation, operation),
                                   Q_ARG(const JSONCallbackParameters&, callbackParams),
                                   Q_ARG(const QByteArray&, dataByteArray),
```
Comment (huffman) [inline libraries/networking/src/AccountManager.cpp]: We should put a return below this call as well. I had meant to fix this in a separate PR.

**Comment by hifi-gustavo** [comment]
Build console output for commit: d9abc79b5d3a7b420071799777bcd553ba5cc87f available at: https://gist.github.com/b1ba2486375b9647d422<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7475.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7475.dmg

**Pull request merged**